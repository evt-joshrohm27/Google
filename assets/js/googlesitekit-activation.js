/**
 * Activation component.
 *
 * This JavaScript loads on every admin page. Reserved for later.
 *
 * Site Kit by Google, Copyright 2021 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * WordPress dependencies
 */
import domReady from '@wordpress/dom-ready';
import { render } from '@wordpress/element';

/**
 * Internal dependencies
 */
import { ActivationApp } from './components/activation/activation-app';
import { VIEW_CONTEXT_ACTIVATION } from './googlesitekit/constants';
import Root from './components/Root';

domReady( () => {
	const renderTarget = document.getElementById(
		'js-googlesitekit-activation'
	);

	if ( renderTarget ) {
		render(
			<Root viewContext={ VIEW_CONTEXT_ACTIVATION }>
				<ActivationApp />
			</Root>,
			renderTarget
		);

		renderTarget.classList.remove( 'googlesitekit-activation--loading' );
	}
} );
