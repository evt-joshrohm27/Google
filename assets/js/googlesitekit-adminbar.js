/**
 * Adminbar component.
 *
 * Site Kit by Google, Copyright 2021 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * External dependencies
 */
import { once } from 'lodash';

/**
 * WordPress dependencies
 */
import domReady from '@wordpress/dom-ready';
import { render } from '@wordpress/element';

/**
 * Internal dependencies.
 */
import { trackEvent } from './util';
import Root from './components/Root';
import AdminBarApp from './components/adminbar/AdminBarApp';
import { VIEW_CONTEXT_ADMIN_BAR } from './googlesitekit/constants';

// Initialize the whole adminbar app.
const init = once( () => {
	const renderTarget = document.getElementById(
		'js-googlesitekit-adminbar-modules'
	);

	if ( renderTarget ) {
		render(
			<Root viewContext={ VIEW_CONTEXT_ADMIN_BAR }>
				<AdminBarApp />
			</Root>,
			renderTarget
		);

		trackEvent( VIEW_CONTEXT_ADMIN_BAR, 'view_urlsummary' );
	}
} );

domReady( () => {
	const siteKitMenuItemEl = document.getElementById(
		'wp-admin-bar-google-site-kit'
	);

	if ( ! siteKitMenuItemEl ) {
		return;
	}

	siteKitMenuItemEl.addEventListener( 'mouseover', init, { once: true } );
	siteKitMenuItemEl.addEventListener( 'focusin', init, { once: true } );
} );
