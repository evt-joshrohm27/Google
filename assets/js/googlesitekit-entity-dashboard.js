/**
 * DashboardDetails component.
 *
 * Site Kit by Google, Copyright 2021 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * WordPress dependencies
 */
import domReady from '@wordpress/dom-ready';
import { render } from '@wordpress/element';

/**
 * Internal dependencies.
 */
import DashboardEntityApp from './components/DashboardEntityApp';
import Root from './components/Root';
import {
	VIEW_CONTEXT_ENTITY_DASHBOARD,
	VIEW_CONTEXT_ENTITY_DASHBOARD_VIEW_ONLY,
} from './googlesitekit/constants';

// Initialize the app once the DOM is ready.
domReady( () => {
	const renderTarget = document.getElementById(
		'js-googlesitekit-entity-dashboard'
	);

	if ( renderTarget ) {
		const { viewOnly } = renderTarget.dataset;

		render(
			<Root
				viewContext={
					viewOnly
						? VIEW_CONTEXT_ENTITY_DASHBOARD_VIEW_ONLY
						: VIEW_CONTEXT_ENTITY_DASHBOARD
				}
			>
				<DashboardEntityApp />
			</Root>,
			renderTarget
		);
	}
} );
