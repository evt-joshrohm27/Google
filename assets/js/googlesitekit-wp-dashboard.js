/**
 * WPDashboard component.
 *
 * Site Kit by Google, Copyright 2021 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
/* eslint camelcase:[0] */

/**
 * WordPress dependencies
 */
import domReady from '@wordpress/dom-ready';
import { render } from '@wordpress/element';

/**
 * Internal dependencies
 */
import Root from './components/Root';
import { VIEW_CONTEXT_WP_DASHBOARD } from './googlesitekit/constants';
import WPDashboardApp from './components/wp-dashboard/WPDashboardApp';

// Initialize the app once the DOM is ready.
domReady( () => {
	const renderTarget = document.getElementById(
		'js-googlesitekit-wp-dashboard'
	);

	if ( renderTarget ) {
		render(
			<Root viewContext={ VIEW_CONTEXT_WP_DASHBOARD }>
				<WPDashboardApp />
			</Root>,
			renderTarget
		);
	}
} );
