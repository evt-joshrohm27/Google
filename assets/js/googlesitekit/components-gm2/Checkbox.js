/**
 * Checkbox component.
 *
 * Site Kit by Google, Copyright 2021 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * External dependencies
 */
import classnames from 'classnames';
import PropTypes from 'prop-types';

/**
 * WordPress dependencies
 */
import { Fragment } from '@wordpress/element';

/**
 * Internal dependencies
 */
import Spinner from '../../components/Spinner';

export default function Checkbox( props ) {
	const {
		onChange,
		id,
		name,
		value,
		checked,
		disabled,
		children,
		tabIndex,
		onKeyDown,
		loading,
	} = props;

	return (
		<div className="mdc-form-field">
			<div
				className={ classnames( 'mdc-checkbox', {
					'mdc-checkbox--disabled': disabled,
				} ) }
			>
				{ loading ? (
					<Spinner isSaving style={ { margin: '0' } } />
				) : (
					<Fragment>
						<input
							className="mdc-checkbox__native-control"
							type="checkbox"
							id={ id }
							name={ name }
							value={ value }
							checked={ checked }
							disabled={ disabled }
							onChange={ onChange }
							tabIndex={ tabIndex }
							onKeyDown={ onKeyDown }
						/>

						<div className="mdc-checkbox__background">
							<svg
								className="mdc-checkbox__checkmark"
								viewBox="0 0 24 24"
							>
								<path
									className="mdc-checkbox__checkmark-path"
									fill="none"
									d="M1.73,12.91 8.1,19.28 22.79,4.59"
								/>
							</svg>
							<div className="mdc-checkbox__mixedmark" />
						</div>
					</Fragment>
				) }
			</div>

			<label htmlFor={ id }>{ children }</label>
		</div>
	);
}

Checkbox.propTypes = {
	onChange: PropTypes.func.isRequired,
	onKeyDown: PropTypes.func,
	id: PropTypes.string.isRequired,
	name: PropTypes.string.isRequired,
	value: PropTypes.string.isRequired,
	checked: PropTypes.bool,
	disabled: PropTypes.bool,
	children: PropTypes.node.isRequired,
	tabIndex: PropTypes.oneOfType( [ PropTypes.number, PropTypes.string ] ),
	loading: PropTypes.bool,
};

Checkbox.defaultProps = {
	checked: false,
	disabled: false,
	tabIndex: undefined,
	onKeyDown: null,
	loading: false,
};
