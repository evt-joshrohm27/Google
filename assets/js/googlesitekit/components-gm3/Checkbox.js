/**
 * Checkbox component.
 *
 * Site Kit by Google, Copyright 2022 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * External dependencies
 */
import '@material/web/checkbox/checkbox';
import PropTypes from 'prop-types';

/**
 * WordPress dependencies
 */
import { useEffect, useRef } from '@wordpress/element';
import Spinner from '../../components/Spinner';

/**
 * Internal dependencies
 */
import { getLabelFromChildren } from '../../util/react-children-to-text';

export default function Checkbox( {
	onChange,
	id,
	name,
	value,
	checked,
	disabled,
	children,
	tabIndex,
	onKeyDown,
	loading,
} ) {
	const ref = useRef( null );

	useEffect( () => {
		const { current } = ref;

		if ( ! current ) {
			return;
		}

		function updateCheckedState() {
			current.checked = checked;

			// The checked property doesn't get reflected to the inner input element checked state,
			// so we need to set it manually.
			const innerInput = current.shadowRoot.querySelector( 'input' );
			if ( innerInput ) {
				innerInput.checked = checked;
			}
		}

		function handleChange( event ) {
			onChange?.( event );

			// Restore current checked state for controlled component behaviour.
			updateCheckedState();
		}

		current.addEventListener( 'change', handleChange );

		if ( onKeyDown ) {
			current.addEventListener( 'keydown', onKeyDown );
		}

		updateCheckedState();

		return () => {
			current.removeEventListener( 'change', handleChange );

			if ( onKeyDown ) {
				current.removeEventListener( 'keydown', onKeyDown );
			}
		};
	}, [ checked, onChange, onKeyDown ] );

	// TODO: Remove `onLabelClick` once the `md-checkbox` component is updated to be a form-associated custom element.
	const onLabelClick = () => {
		const { current } = ref;

		if ( ! current || disabled ) {
			return;
		}

		current.checked = ! current.checked;

		current.dispatchEvent(
			new Event( 'change', { bubbles: true, cancelable: true } )
		);
	};

	return (
		<div className="googlesitekit-component-gm3_checkbox">
			{ loading && (
				<div className="googlesitekit-component-gm3_checkbox--loading">
					<Spinner isSaving />
				</div>
			) }
			{ ! loading && (
				<md-checkbox
					id={ id }
					ref={ ref }
					role="checkbox"
					aria-label={ getLabelFromChildren( children ) }
					// `Lit` boolean attributes treat anything non-null/undefined as true.
					// Coerce to null if false.
					// See https://lit.dev/docs/v1/components/properties/#attributes
					checked={ checked || null }
					disabled={ disabled || null }
					name={ name }
					value={ value }
					tabIndex={ tabIndex }
				></md-checkbox>
			) }
			{
				// These ESLint rules are disabled because we are intentionally adding a click handler to the
				// label to make up for a bug in the `@material/web` component, where the label does not respond to
				// click events for the checkbox. Usually, labels associated with inputs don't respond to
				// keyboard events, but due to our hack this is required.
				/* eslint-disable jsx-a11y/click-events-have-key-events */
				/* eslint-disable jsx-a11y/no-noninteractive-element-interactions */
			 }
			<label htmlFor={ id } onClick={ onLabelClick }>
				{ children }
			</label>
			{ /* eslint-enable jsx-a11y/click-events-have-key-events */
			/* eslint-enable jsx-a11y/no-noninteractive-element-interactions */ }
		</div>
	);
}

Checkbox.propTypes = {
	onChange: PropTypes.func.isRequired,
	onKeyDown: PropTypes.func,
	id: PropTypes.string.isRequired,
	name: PropTypes.string.isRequired,
	value: PropTypes.string.isRequired,
	checked: PropTypes.bool,
	disabled: PropTypes.bool,
	children: PropTypes.node.isRequired,
	tabIndex: PropTypes.oneOfType( [ PropTypes.number, PropTypes.string ] ),
	loading: PropTypes.bool,
};

Checkbox.defaultProps = {
	checked: false,
	disabled: false,
	tabIndex: undefined,
	onKeyDown: null,
	loading: false,
};
