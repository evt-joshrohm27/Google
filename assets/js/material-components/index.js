/**
 * Material components.
 *
 * Site Kit by Google, Copyright 2021 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * External dependencies
 */
import TextField, { HelperText, Input } from '@material/react-text-field';
import { MDCRipple } from '@material/ripple';
import { MDCFormField } from '@material/form-field';
import { MDCRadio } from '@material/radio';
import Select, { Option } from '@material/react-select';
import { MDCSwitch } from '@material/switch';
import { MDCDialog } from '@material/dialog';
import { MDCCheckbox } from '@material/checkbox';
import { MDCMenu } from '@material/menu';
import Dialog, { DialogContent, DialogFooter } from '@material/react-dialog';
import { CircularProgress } from '@material-ui/core';

export * from './layout';
export {
	TextField,
	HelperText,
	Input,
	MDCRipple,
	MDCFormField,
	MDCRadio,
	Select,
	Option,
	MDCSwitch,
	MDCDialog,
	MDCCheckbox,
	MDCMenu,
	Dialog,
	DialogContent,
	DialogFooter,
	CircularProgress,
};
