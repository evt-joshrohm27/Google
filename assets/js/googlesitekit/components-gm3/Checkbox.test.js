/**
 * Checkbox tests.
 *
 * Site Kit by Google, Copyright 2022 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * WordPress dependencies
 */
import { useState } from '@wordpress/element';

/**
 * Internal dependencies
 */
import { fireEvent, render } from '../../../../tests/js/test-utils';
import Checkbox from './Checkbox';

describe( 'Checkbox', () => {
	it( 'should render the checkbox', () => {
		const { container } = render(
			<Checkbox
				id="checkbox-id"
				name="checkbox-name"
				value="checkbox-value"
				onChange={ () => {} }
			>
				Checkbox Label
			</Checkbox>
		);
		expect( container ).toMatchSnapshot();
	} );

	it( 'should render the checkbox with checked state', () => {
		const { container } = render(
			<Checkbox
				id="checkbox-id"
				name="checkbox-name"
				value="checkbox-value"
				onChange={ () => {} }
				checked
			>
				Checkbox Label
			</Checkbox>
		);
		expect( container ).toMatchSnapshot();
	} );

	it( 'should render the checkbox with disabled state', () => {
		const { container } = render(
			<Checkbox
				id="checkbox-id"
				name="checkbox-name"
				value="checkbox-value"
				onChange={ () => {} }
				disabled
			>
				Checkbox Label
			</Checkbox>
		);
		expect( container ).toMatchSnapshot();
	} );

	it( 'should render the checkbox with loading state', () => {
		const { container } = render(
			<Checkbox
				id="checkbox-id"
				name="checkbox-name"
				value="checkbox-value"
				onChange={ () => {} }
				loading
			>
				Checkbox Label
			</Checkbox>
		);
		expect( container ).toMatchSnapshot();
	} );

	it( 'should render the checkbox with a complex label', () => {
		const { container } = render(
			<Checkbox
				id="checkbox-id"
				name="checkbox-name"
				value="checkbox-value"
				onChange={ () => {} }
			>
				<div>
					Complex Label
					<span>
						<a href="https://example.com"> With</a>
						<span>{ 5 }</span>
					</span>
					<span>
						Sub<span>&nbsp;Children </span>
					</span>
				</div>
			</Checkbox>
		);
		expect( container ).toMatchSnapshot();
		expect(
			container
				.querySelector( 'md-checkbox' )
				.getAttribute( 'data-aria-label' )
		).toBe( 'Complex Label With 5 Sub Children' );
	} );

	it( 'should attach the onKeyDown handler when present', () => {
		const onKeyDown = jest.fn();

		const { getByRole } = render(
			<Checkbox
				id="checkbox-id"
				name="checkbox-name"
				value="checkbox-value"
				onChange={ () => {} }
				onKeyDown={ onKeyDown }
			>
				Checkbox Label
			</Checkbox>
		);

		fireEvent.keyDown( getByRole( 'checkbox' ), {
			key: 'Enter',
			keyCode: 13,
		} );

		expect( onKeyDown ).toHaveBeenCalledTimes( 1 );
		expect( onKeyDown ).toHaveBeenCalledWith(
			expect.objectContaining( {
				key: 'Enter',
				keyCode: 13,
			} )
		);
	} );

	describe.each( [
		[
			// Clickable element for test description:
			'input',
			// Function to retrieve the clickable element:
			( { getByRole } ) =>
				getByRole( 'checkbox' ).shadowRoot.querySelector( 'input' ),
		],
		[
			// Clickable element for test description:
			'label',
			// Function to retrieve the clickable element:
			( { getByText } ) => getByText( 'Checkbox Label' ),
		],
	] )(
		'controlled input behaviour for %s',
		( clickableElement, getClickableElement ) => {
			function createOnChangeSpy() {
				// We need to explicitly track the `checked` state at the time the `onChange` handler was called.
				// It's not sufficient to simply check `onChange.mock.calls[...]`, as `event.target.checked` may have
				// changed by the time the value is examined.
				const onChangeCalls = [];
				const onChange = jest.fn( ( event ) => {
					onChangeCalls.push( { checked: event.target.checked } );
				} );
				return { onChange, onChangeCalls };
			}

			function expectCheckboxToBeChecked( checkbox ) {
				expect( checkbox ).toHaveAttribute( 'checked' );

				// Explicitly check the `checked` attribute of the underlying input element. This is
				// worthwhile as we are explicitly reaching into the shadow DOM to update the input within
				// the Checkbox component.
				expect(
					checkbox.shadowRoot.querySelector( 'input' )
				).toHaveAttribute( 'checked' );
			}

			function expectCheckboxNotToBeChecked( checkbox ) {
				expect( checkbox ).not.toHaveAttribute( 'checked' );
				expect(
					checkbox.shadowRoot.querySelector( 'input' )
				).not.toHaveAttribute( 'checked' );
			}

			it( 'should correctly invoke onChange and retain its unchecked state when clicked', async () => {
				const { onChange, onChangeCalls } = createOnChangeSpy();
				const result = render(
					<Checkbox
						id="checkbox-id"
						name="checkbox-name"
						value="checkbox-value"
						onChange={ onChange }
					>
						Checkbox Label
					</Checkbox>
				);

				const { container, getByRole } = result;

				await getByRole( 'checkbox' ).updateComplete;

				expect( container ).toMatchSnapshot();

				// Confirm the checkbox is not checked.
				expectCheckboxNotToBeChecked( getByRole( 'checkbox' ) );

				fireEvent.click( getClickableElement( result ) );

				await getByRole( 'checkbox' ).updateComplete;

				expect( container ).toMatchSnapshot();

				// Confirm the click resulted in a change event with target.checked: true.
				expect( onChange ).toHaveBeenCalledTimes( 1 );
				expect( onChangeCalls[ 0 ].checked ).toBe( true );

				// Confirm the checkbox remains unchecked as its state is controlled.
				expectCheckboxNotToBeChecked( getByRole( 'checkbox' ) );
			} );

			it( 'should correctly invoke onChange and retain its checked state when clicked', async () => {
				const { onChange, onChangeCalls } = createOnChangeSpy();
				const result = render(
					<Checkbox
						id="checkbox-id"
						name="checkbox-name"
						value="checkbox-value"
						checked
						onChange={ onChange }
					>
						Checkbox Label
					</Checkbox>
				);

				const { container, getByRole } = result;

				await getByRole( 'checkbox' ).updateComplete;

				expect( container ).toMatchSnapshot();

				// Confirm the checkbox is checked.
				expectCheckboxToBeChecked( getByRole( 'checkbox' ) );

				fireEvent.click( getClickableElement( result ) );

				await getByRole( 'checkbox' ).updateComplete;

				expect( container ).toMatchSnapshot();

				// Confirm the click resulted in a change event with target.checked: false.
				expect( onChange ).toHaveBeenCalledTimes( 1 );
				expect( onChangeCalls[ 0 ].checked ).toBe( false );

				// Confirm the checkbox remains checked as its state is controlled.
				expectCheckboxToBeChecked( getByRole( 'checkbox' ) );
			} );

			it( 'should allow updating of the checked state', async () => {
				function CheckableCheckbox( { onChange } ) {
					const [ checked, setChecked ] = useState( false );

					return (
						<Checkbox
							id="checkbox-id"
							name="checkbox-name"
							value="checkbox-value"
							checked={ checked }
							onChange={ ( event ) => {
								onChange( event );
								setChecked( event.target.checked );
							} }
						>
							Checkbox Label
						</Checkbox>
					);
				}

				const { onChange, onChangeCalls } = createOnChangeSpy();
				const result = render(
					<CheckableCheckbox onChange={ onChange } />
				);

				const { container, getByRole } = result;

				await getByRole( 'checkbox' ).updateComplete;

				expect( container ).toMatchSnapshot();

				// Confirm the checkbox is not checked.
				expectCheckboxNotToBeChecked( getByRole( 'checkbox' ) );

				fireEvent.click( getClickableElement( result ) );

				await getByRole( 'checkbox' ).updateComplete;

				expect( container ).toMatchSnapshot();

				// Confirm the click resulted in a change event with target.checked: true.
				expect( onChange ).toHaveBeenCalledTimes( 1 );
				expect( onChangeCalls[ 0 ].checked ).toBe( true );

				// Confirm the checkbox is now checked
				expectCheckboxToBeChecked( getByRole( 'checkbox' ) );

				// Click again to uncheck
				onChangeCalls.length = 0;
				onChange.mockClear();

				fireEvent.click( getClickableElement( result ) );

				await getByRole( 'checkbox' ).updateComplete;

				expect( container ).toMatchSnapshot();

				// Confirm the click resulted in a change event with target.checked: false.
				expect( onChange ).toHaveBeenCalledTimes( 1 );
				expect( onChangeCalls[ 0 ].checked ).toBe( false );

				// Confirm the checkbox is now unchecked
				expectCheckboxNotToBeChecked( getByRole( 'checkbox' ) );
			} );
		}
	);
} );
