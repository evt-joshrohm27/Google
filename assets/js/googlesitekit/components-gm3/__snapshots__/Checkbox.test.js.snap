// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox controlled input behaviour for input should allow updating of the checked state 1`] = `
<div>
  <div
    class="googlesitekit-component-gm3_checkbox"
  >
    <md-checkbox
      data-aria-label="Checkbox Label"
      id="checkbox-id"
      name="checkbox-name"
      role="checkbox"
      value="checkbox-value"
    />
    <label
      for="checkbox-id"
    >
      Checkbox Label
    </label>
  </div>
</div>
`;

exports[`Checkbox controlled input behaviour for input should allow updating of the checked state 2`] = `
<div>
  <div
    class="googlesitekit-component-gm3_checkbox"
  >
    <md-checkbox
      checked=""
      data-aria-label="Checkbox Label"
      id="checkbox-id"
      name="checkbox-name"
      role="checkbox"
      value="checkbox-value"
    />
    <label
      for="checkbox-id"
    >
      Checkbox Label
    </label>
  </div>
</div>
`;

exports[`Checkbox controlled input behaviour for input should allow updating of the checked state 3`] = `
<div>
  <div
    class="googlesitekit-component-gm3_checkbox"
  >
    <md-checkbox
      data-aria-label="Checkbox Label"
      id="checkbox-id"
      name="checkbox-name"
      role="checkbox"
      value="checkbox-value"
    />
    <label
      for="checkbox-id"
    >
      Checkbox Label
    </label>
  </div>
</div>
`;

exports[`Checkbox controlled input behaviour for input should correctly invoke onChange and retain its checked state when clicked 1`] = `
<div>
  <div
    class="googlesitekit-component-gm3_checkbox"
  >
    <md-checkbox
      checked=""
      data-aria-label="Checkbox Label"
      id="checkbox-id"
      name="checkbox-name"
      role="checkbox"
      value="checkbox-value"
    />
    <label
      for="checkbox-id"
    >
      Checkbox Label
    </label>
  </div>
</div>
`;

exports[`Checkbox controlled input behaviour for input should correctly invoke onChange and retain its checked state when clicked 2`] = `
<div>
  <div
    class="googlesitekit-component-gm3_checkbox"
  >
    <md-checkbox
      checked=""
      data-aria-label="Checkbox Label"
      id="checkbox-id"
      name="checkbox-name"
      role="checkbox"
      value="checkbox-value"
    />
    <label
      for="checkbox-id"
    >
      Checkbox Label
    </label>
  </div>
</div>
`;

exports[`Checkbox controlled input behaviour for input should correctly invoke onChange and retain its unchecked state when clicked 1`] = `
<div>
  <div
    class="googlesitekit-component-gm3_checkbox"
  >
    <md-checkbox
      data-aria-label="Checkbox Label"
      id="checkbox-id"
      name="checkbox-name"
      role="checkbox"
      value="checkbox-value"
    />
    <label
      for="checkbox-id"
    >
      Checkbox Label
    </label>
  </div>
</div>
`;

exports[`Checkbox controlled input behaviour for input should correctly invoke onChange and retain its unchecked state when clicked 2`] = `
<div>
  <div
    class="googlesitekit-component-gm3_checkbox"
  >
    <md-checkbox
      data-aria-label="Checkbox Label"
      id="checkbox-id"
      name="checkbox-name"
      role="checkbox"
      value="checkbox-value"
    />
    <label
      for="checkbox-id"
    >
      Checkbox Label
    </label>
  </div>
</div>
`;

exports[`Checkbox controlled input behaviour for label should allow updating of the checked state 1`] = `
<div>
  <div
    class="googlesitekit-component-gm3_checkbox"
  >
    <md-checkbox
      data-aria-label="Checkbox Label"
      id="checkbox-id"
      name="checkbox-name"
      role="checkbox"
      value="checkbox-value"
    />
    <label
      for="checkbox-id"
    >
      Checkbox Label
    </label>
  </div>
</div>
`;

exports[`Checkbox controlled input behaviour for label should allow updating of the checked state 2`] = `
<div>
  <div
    class="googlesitekit-component-gm3_checkbox"
  >
    <md-checkbox
      checked=""
      data-aria-label="Checkbox Label"
      id="checkbox-id"
      name="checkbox-name"
      role="checkbox"
      value="checkbox-value"
    />
    <label
      for="checkbox-id"
    >
      Checkbox Label
    </label>
  </div>
</div>
`;

exports[`Checkbox controlled input behaviour for label should allow updating of the checked state 3`] = `
<div>
  <div
    class="googlesitekit-component-gm3_checkbox"
  >
    <md-checkbox
      data-aria-label="Checkbox Label"
      id="checkbox-id"
      name="checkbox-name"
      role="checkbox"
      value="checkbox-value"
    />
    <label
      for="checkbox-id"
    >
      Checkbox Label
    </label>
  </div>
</div>
`;

exports[`Checkbox controlled input behaviour for label should correctly invoke onChange and retain its checked state when clicked 1`] = `
<div>
  <div
    class="googlesitekit-component-gm3_checkbox"
  >
    <md-checkbox
      checked=""
      data-aria-label="Checkbox Label"
      id="checkbox-id"
      name="checkbox-name"
      role="checkbox"
      value="checkbox-value"
    />
    <label
      for="checkbox-id"
    >
      Checkbox Label
    </label>
  </div>
</div>
`;

exports[`Checkbox controlled input behaviour for label should correctly invoke onChange and retain its checked state when clicked 2`] = `
<div>
  <div
    class="googlesitekit-component-gm3_checkbox"
  >
    <md-checkbox
      checked=""
      data-aria-label="Checkbox Label"
      id="checkbox-id"
      name="checkbox-name"
      role="checkbox"
      value="checkbox-value"
    />
    <label
      for="checkbox-id"
    >
      Checkbox Label
    </label>
  </div>
</div>
`;

exports[`Checkbox controlled input behaviour for label should correctly invoke onChange and retain its unchecked state when clicked 1`] = `
<div>
  <div
    class="googlesitekit-component-gm3_checkbox"
  >
    <md-checkbox
      data-aria-label="Checkbox Label"
      id="checkbox-id"
      name="checkbox-name"
      role="checkbox"
      value="checkbox-value"
    />
    <label
      for="checkbox-id"
    >
      Checkbox Label
    </label>
  </div>
</div>
`;

exports[`Checkbox controlled input behaviour for label should correctly invoke onChange and retain its unchecked state when clicked 2`] = `
<div>
  <div
    class="googlesitekit-component-gm3_checkbox"
  >
    <md-checkbox
      data-aria-label="Checkbox Label"
      id="checkbox-id"
      name="checkbox-name"
      role="checkbox"
      value="checkbox-value"
    />
    <label
      for="checkbox-id"
    >
      Checkbox Label
    </label>
  </div>
</div>
`;

exports[`Checkbox should render the checkbox 1`] = `
<div>
  <div
    class="googlesitekit-component-gm3_checkbox"
  >
    <md-checkbox
      data-aria-label="Checkbox Label"
      id="checkbox-id"
      name="checkbox-name"
      role="checkbox"
      value="checkbox-value"
    />
    <label
      for="checkbox-id"
    >
      Checkbox Label
    </label>
  </div>
</div>
`;

exports[`Checkbox should render the checkbox with a complex label 1`] = `
<div>
  <div
    class="googlesitekit-component-gm3_checkbox"
  >
    <md-checkbox
      data-aria-label="Complex Label With 5 Sub Children"
      id="checkbox-id"
      name="checkbox-name"
      role="checkbox"
      value="checkbox-value"
    />
    <label
      for="checkbox-id"
    >
      <div>
        Complex Label
        <span>
          <a
            href="https://example.com"
          >
             With
          </a>
          <span>
            5
          </span>
        </span>
        <span>
          Sub
          <span>
            Â Children 
          </span>
        </span>
      </div>
    </label>
  </div>
</div>
`;

exports[`Checkbox should render the checkbox with checked state 1`] = `
<div>
  <div
    class="googlesitekit-component-gm3_checkbox"
  >
    <md-checkbox
      checked="true"
      data-aria-label="Checkbox Label"
      id="checkbox-id"
      name="checkbox-name"
      role="checkbox"
      value="checkbox-value"
    />
    <label
      for="checkbox-id"
    >
      Checkbox Label
    </label>
  </div>
</div>
`;

exports[`Checkbox should render the checkbox with disabled state 1`] = `
<div>
  <div
    class="googlesitekit-component-gm3_checkbox"
  >
    <md-checkbox
      data-aria-label="Checkbox Label"
      disabled="true"
      id="checkbox-id"
      name="checkbox-name"
      role="checkbox"
      value="checkbox-value"
    />
    <label
      for="checkbox-id"
    >
      Checkbox Label
    </label>
  </div>
</div>
`;

exports[`Checkbox should render the checkbox with loading state 1`] = `
<div>
  <div
    class="googlesitekit-component-gm3_checkbox"
  >
    <div
      class="googlesitekit-component-gm3_checkbox--loading"
    >
      <span
        class="spinner"
        style="display: inline-block; float: none; margin-top: 0px; visibility: visible;"
      />
    </div>
    <label
      for="checkbox-id"
    >
      Checkbox Label
    </label>
  </div>
</div>
`;
